core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -85,21 +86,43 @@ public ValidationResponse validateChallenge(ValidationRequest request)
     ChallengeTransaction challenge = parseChallenge(request);
     // pre validation to be defined by the anchor
     preValidateRequestValidation(request, challenge);
+
+    String homeDomain = validateChallengeTransactionHomeDomain(challenge);
+
     // fetch the client domain from the transaction
     String clientDomain = fetchClientDomain(challenge);
     // fetch the account response from the horizon
     AccountResponse account = fetchAccount(request, challenge, clientDomain);
 
     if (account == null) {
       // The account does not exist from Horizon, using the client's master key to verify.
-      return ValidationResponse.of(generateSep10Jwt(challenge, clientDomain));
+      return ValidationResponse.of(generateSep10Jwt(challenge, clientDomain, homeDomain));
     }
     // Since the account exists, we should check the signers and the client domain
     validateChallengeRequest(request, account, clientDomain);
     // increment counter
     incrementValidationRequestValidatedCounter();
     // Generate the JWT token
-    return ValidationResponse.of(generateSep10Jwt(challenge, clientDomain));
+    return ValidationResponse.of(generateSep10Jwt(challenge, clientDomain, homeDomain));
+  }
+
+  @Override
+  public String validateChallengeTransactionHomeDomain(ChallengeTransaction challenge)
+      throws SepValidationException {
+    String homeDomain;
+    try {
+      ManageDataOperation operation =
+          (ManageDataOperation) challenge.getTransaction().getOperations()[0];
+      homeDomain = operation.getName().split(" ")[0];
```

Comment:
Can this be replaced by `challenge.getMatchedHomeDomain()`?

Reply:
Why not use the one from the SDK? https://stellar.github.io/java-stellar-sdk/org/stellar/sdk/Sep10Challenge.ChallengeTransaction.html#getMatchedHomeDomain()